mofa/model.py

```diff
@@ -155,6 +156,72 @@ class LigandDescription:
     def atoms(self):
         return read_from_string(self.xyz, "xyz")
 
+    def anchor_constrained_optimization(self, xyz_tol=0.001, force_constant=10000.0, max_iterations=1000):
+        """optimize the ligand while the anchor atoms are constrained
+
+        Args:
+            xyz_tol: coordinate tolerance for constrained atoms
+            force_constant: the spring force constant to keep the constrained atoms in original position
+            max_iterations: maximum number of iterations for optimization
+        Returns:
+            inplace function, no return
+        """
+
+        mol = Chem.MolFromXYZBlock(self.xyz)
+        all_anchor_atoms = list(itertools.chain(*self.anchor_atoms))
+        if self.anchor_type != "COO" and self.dummy_element != "At" and len(self.anchor_atoms[0]) != 3:
+            charge = int(0)
+        else:
+            charge = int(-len(self.anchor_atoms))
+        rdDetermineBonds.DetermineBonds(mol, charge=charge)
+        rdDetermineBonds.DetermineConnectivity(mol)
+        rdDetermineBonds.DetermineBondOrders(mol)
+        molprop = AllChem.MMFFGetMoleculeProperties(mol)
+        ff = AllChem.MMFFGetMoleculeForceField(mol, molprop)
+        # set constraints
+        for i in all_anchor_atoms:
+            ff.MMFFAddPositionConstraint(i, xyz_tol, force_constant)
+        ff.Minimize(maxIts=max_iterations)
+        self.xyz = Chem.MolToXYZBlock(mol)
+
+    def swap_cyano_with_COO(self):
+        """create a new LigandDescription object with the same middle part but with the -COO instead of -cyano groups
+
+        Returns:
+            the new -COO LigandDescription object
+        """
+
+        if self.dummy_element != "Fr" or self.anchor_type == "COO":
```

Comment:
If this is an error condition, raise a `ValueError` to say so rather than `None`.